app/ui-react/syndesis/src/modules/integrations/pages/detail/DetailsPage.tsx

```diff
@@ -22,18 +74,237 @@ export interface IDetailsPageState {
+export class DetailsPage extends React.Component<
+  IIntegrationDetailsPageProps,
+  IIntegrationDetailsPageState
+> {
+  public constructor(props: IIntegrationDetailsPageProps) {
+    super(props);
+    this.state = {
+      showPromptDialog: false,
+    };
+    this.handleAction = this.handleAction.bind(this);
+    this.handleActionCancel = this.handleActionCancel.bind(this);
+    this.promptForAction = this.promptForAction.bind(this);
+  }
+
+  public handleActionCancel() {
+    this.setState({
+      showPromptDialog: false,
+    });
+  }
+
+  public handleAction() {
+    const action = this.state.handleAction;
+    this.setState({
+      showPromptDialog: false,
+    });
+    if (typeof action === 'function') {
+      action.apply(this);
+    } else {
+      throw Error('Undefined action set for confirmation dialog');
+    }
+  }
+
+  public promptForAction(options: IPromptActionOptions) {
+    this.setState({
+      ...options,
+      showPromptDialog: true,
+    });
+  }
+
   public render() {
     return (
-      <WithRouteData<IDetailsPageParams, IDetailsPageState>>
-        {({ integrationId }, { integration }, { history }) => {
+      <WithRouteData<IIntegrationDetailsRouteParams, null>>
+        {({ integrationId }) => {
           return (
-            <div>
-              <Translation ns={['integration', 'shared']}>
-                {t => <IntegrationDetailNavBar integration={integration} />}
-              </Translation>
-              <p>This is the Integration Detail History page.</p>
-            </div>
+            <WithIntegrationHelpers>
+              {({
+                deleteIntegration,
+                deployIntegration,
+                exportIntegration,
+                undeployIntegration,
+              }) => {
+                return (
+                  <WithIntegration integrationId={integrationId}>
+                    {({ data, hasData, error }) => (
+                      <WithLoader
+                        error={error}
+                        loading={!hasData}
+                        loaderChildren={<Loader />}
+                        errorChildren={<div>TODO</div>}
+                      >
+                        {() => (
+                          <div>
+                            <Translation ns={['integrations', 'shared']}>
+                              {t => {
+                                const deployments = data.deployments
+                                  ? data.deployments
+                                  : [];
+                                const editAction: IIntegrationAction = {
+                                  href: resolvers.integration.edit.index({
+                                    flowId: '0',
+                                    integration: data,
+                                  }),
+                                  label: 'Edit',
+                                };
+                                const startAction: IIntegrationAction = {
```

Comment:
Maybe open an issue to refactor this in a later moment? I really don't like how this is repeated somewhere else in the codebase.